clang/test/CIR/CodeGen/tbaa-struct.cpp

```diff
@@ -0,0 +1,34 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir -O1
+// RUN: FileCheck --check-prefix=CIR --input-file=%t.cir %s
+
+// CIR: #tbaa[[tbaa_NYI:.*]] = #cir.tbaa
+// CIR: #tbaa[[INT:.*]] = #cir.tbaa_scalar<type = !u32i>
+// CIR: #tbaa[[INT_PTR:.*]] = #cir.tbaa_scalar<type = !cir.ptr<!u32i>>
+// CIR: #tbaa[[StructA_PTR:.*]] = #cir.tbaa_scalar<type = !cir.ptr<!ty_StructA>>
```

Comment:
I'm a bit confused here, why can a tbaa_scalar have a struct type? This also doesn't seem to be used anywhere.

Reply:
it is a pointer to struct `StructA`. 
added missing check